qt/python/mantidqt/mantidqt/widgets/rawdataexplorer/model.py

```diff
@@ -0,0 +1,305 @@
+        """
+        super().__init__()
+        self.presenter: "RawDataExplorerPresenter" = presenter
+
+        self._previews: list = list()
+
+        self.memory_manager: MemoryManager = MemoryManager(self)
+
+    def modify_preview(self, filename: str):
+        """
+        Modify the last preview of the same type. If none is found, a new preview is opened.
+        @param filename: the full path to the file to show
+        """
+
+        ws_name = Path(filename).stem
+
+        # load the file if necessary
+        if not mtd.doesExist(ws_name):
+            success = self.load_file(ws_name, filename)
+            if not success:
+                return
+
+        # notify the memory manager about this interaction
+        self.memory_manager.workspace_interacted_with(ws_name)
+
+        # determine the preview given the prepared data
+        preview = self.choose_preview(ws_name)
+
+        if preview is None:
+            return
+
+        # check the previews currently opened in case one of the correct type already exists and can be replaced.
+        # NB: currently, previews only has one element max at a given time, but that could change so the design stays.
+        for current_preview in reversed(self._previews):
+            if current_preview.get_preview_type() == preview:
+                if self.presenter.is_accumulating():
+                    ws_name = self.accumulate(current_preview.get_workspace_name(), ws_name)
+                    if ws_name is None:
+                        return
+                    self.memory_manager.workspace_interacted_with(ws_name)
+                current_preview.set_workspace_name(ws_name)
+                return
+
+        # create a new preview, since none corresponding were found
+        preview_model = PreviewModel(preview, ws_name)
+
+        # close all already existing previews
+        previews = [preview for preview in self._previews]
+        for preview in previews:
+            preview.sig_request_close.emit()
+            self.del_preview(preview)
+
+        # add this new preview
+        self._previews.append(preview_model)
+        self.sig_new_preview.emit(preview_model)
+
+    @staticmethod
+    def load_file(workspace_name: str, filename: str) -> bool:
+        """
+        Load the workspace to the workspace name.
+        @param workspace_name: the name of the workspace to create.
+        @param filename: the path to the file to load.
+        @return whether the loading succeeded.
+        """
+        load_alg = AlgorithmManager.create("Load")
+        load_alg.setLogging(True)
+        load_alg.setProperty("OutputWorkspace", workspace_name)
+        try:
+            load_alg.setProperty("Filename", filename)
+            load_alg.execute()
+        except RuntimeError as e:
+            error_reporting("Error, could not load file:\n {0}".format(e))
+            return False
+
+        if not load_alg.isExecuted():
```

Comment:
I don't think it is. L167 checks that the algorithm hasn't thrown an error, but I can imagine cases where no error is thrown yet the algorithm failed to execute. Is it possible for any loader ? I don't know, but I don't know otherwise either.